Lib/shutil.py

```diff
@@ -442,7 +442,7 @@ def _ignore_patterns(path, names):
 def _copytree(entries, src, dst, symlinks, ignore, copy_function,
               ignore_dangling_symlinks, dirs_exist_ok=False):
     if ignore is not None:
-        ignored_names = ignore(src, set(os.listdir(src)))
+        ignored_names = ignore(src, set([x.name for x in entries]))
```

Comment:
```suggestion
        ignored_names = ignore(src, {x.name for x in entries}))
```

Reply:
:+1: thanks @sir-sigurd !

Reply:
(Hope it's OK to merge the commit suggestion, or should it be rebased/squashed?)

Reply:
```
 File "/home/travis/build/python/cpython/Lib/shutil.py", line 445

    ignored_names = ignore(src, {x.name for x in entries}))

                                                          ^

SyntaxError: unmatched ')'

Makefile:614: recipe for target 'sharedmods' failed

make: *** [sharedmods] Error 1
```

Reply:
Thanks @giampaolo, fixed, then squashed commits. Also rebased on `master` again :+1: